libibverbs/init.c

```diff
@@ -486,6 +486,7 @@ int ibverbs_get_device_list(struct list_head *device_list)
 				"Warning: no userspace device-specific driver found for %s\n",
 				sysfs_dev->sysfs_path);
 		}
+		list_del(&sysfs_dev->entry);
```

Comment:
This is useless, the sysfs_list is on stack, and sysfs_dev->entry points to that on-stack list. There is no actual reason to list_del it when the sysfs_list will evaporate on funciton exit.

Reply:
@jgunthorpe : get it.